registry/auth/token/token.go

```diff
@@ -144,8 +147,14 @@ func (t *Token) Verify(verifyOpts VerifyOptions) error {
 
 	// Verify that the token is currently usable and not expired.
 	currentUnixTime := time.Now().Unix()
-	if !(t.Claims.NotBefore <= currentUnixTime && currentUnixTime <= t.Claims.Expiration) {
-		log.Errorf("token not to be used before %d or after %d - currently %d", t.Claims.NotBefore, t.Claims.Expiration, currentUnixTime)
+
+	if currentUnixTime >= t.Claims.Expiration+int64(Leeway.Seconds()) {
+		log.Errorf("token not to be used after %d - currently %d", t.Claims.Expiration, currentUnixTime)
+		return ErrInvalidToken
```

Comment:
A user manually inspecting the token would be able to work out it's expired so it doesn't seem useful to be opaque about it. If it's easy to make the error message more explicit I'd say we should do it.